umbral/_pre.py

```diff
@@ -131,22 +135,24 @@ def assess_cfrag_correctness(cfrag: 'CapsuleFrag', capsule: 'Capsule') -> bool:
 
 
 def verify_kfrag(kfrag: 'KFrag',
-                 delegating_pubkey: UmbralPublicKey,
-                 signing_pubkey: UmbralPublicKey,
-                 receiving_pubkey: UmbralPublicKey
+                 params: UmbralParameters,
+                 delegating_pubkey: Optional[UmbralPublicKey] = None,
+                 signing_pubkey: Optional[UmbralPublicKey] = None,
+                 receiving_pubkey: Optional[UmbralPublicKey] = None,
                  ) -> bool:
 
+    if params is None:
+        params = default_params()
 
-    params = delegating_pubkey.params
-    if not params == receiving_pubkey.params:
-        raise ValueError("The delegating and receiving keys must use the same UmbralParameters")
+    if delegating_pubkey and delegating_pubkey.params != params:
+            raise ValueError("The delegating key uses different UmbralParameters")
 
-    u = params.u
+    if receiving_pubkey and receiving_pubkey.params != params:
+            raise ValueError("The receiving key uses different UmbralParameters")
 
-    delegating_point = delegating_pubkey.point_key
-    receiving_point = receiving_pubkey.point_key
+    u = params.u
 
-    id = kfrag._id
+    kfrag_id = kfrag._id
```

Comment:
Certainly a step in the right direction.  :-)

But again: doesn't this suggest that `kfrag._id` wants to be public?

Reply:
In a way, yes ... although I kind of recall a consensus on that "_this is fine as long as we are calling private attributes from a private module_" (in this case`umbral._pre`). My memory can be wrong on this, though.

The problem is that I've just checked and there are several private attributes of `KFrag` (including `_id`) that are called in `umbral.pre`, so I guess I'll have to make these public. But I think these PR may not be the place, because it's not only `KFrag`, this is happening all over the place. 

And by the way, perhaps unrelated, I don't like the `umbral._pre` module. Ideally, I'd want to remove it once we assess the situation with the right attributes visibility.